jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcStorageHandler.java

```diff
@@ -99,7 +101,10 @@ public void configureInputJobProperties(TableDesc tableDesc, Map<String, String>
   @Override
   public URI getURIForAuth(Table table) throws URISyntaxException {
     Map<String, String> tableProperties = HiveCustomStorageHandlerUtils.getTableProperties(table);
-    String host_url = tableProperties.get(Constants.JDBC_URL);
+    DatabaseType dbType = DatabaseType.valueOf(
+      tableProperties.get(JdbcStorageConfig.DATABASE_TYPE.getPropertyName()));
+    String host_url = DatabaseType.METASTORE == dbType ?
+      "internal:metastore://" : tableProperties.get(Constants.JDBC_URL);
```

Comment:
All the storage handlers will have prefix in the Storage URL including jdbc storage handler. Can we make this hardcoded uri also to be in the same format? Essentially you would need to add "JDBC://" before the string "internal:metastore:/" (please take out extra "/" after "metastore:/")?

Reply:
With the current setup
````
'HivePrivilegeObject':{'type':STORAGEHANDLER_URI, 'dbName':null, 'objectType':STORAGEHANDLER_URI, 'objectName':internal:metastore:///null, 'columns':[], 'partKeys':[], 'commandParams':[], 'actionType':OTHER, 'owner':hive}
````
elements={storage-type=internal:metastore; storage-url=/null; }

Why do we need to remove extra "/" after "metastore?

Reply:
should it be `jdbc:metastore://`?

Reply:
Yeah, having `jdbc://internal:metastore:/` would be consistent with other Storage URIs. But I'm ok with your above suggestion as well.

Reply:
updated, please take a look